fairlearn/metrics/_disparities.py

```diff
@@ -7,6 +7,84 @@
 from ._metric_frame import MetricFrame
 
 
+def average_odds_difference(
```

Comment:
I'm fine including the two metrics--the goal and even behavior is very similar to our existing `equalized_odds_difference` and `equalized_odds_ratio`, in the sense that `average_odds_difference==0` or `average_odds_ratio==1` indicate that there is no disparity according to the equalized odds criterion.

But I think that the name is confusing for two reasons:
* the word "odds" is in statistics [defined](https://en.wikipedia.org/wiki/Odds#Statistical_usage) as a ratio of "probability of event / 1-probability of event" and we are definitely not averaging such odds...
* it turns out that [AIF360's average_odds_difference](https://aif360.readthedocs.io/en/latest/modules/generated/aif360.sklearn.metrics.average_odds_difference.html) is calculating something different than this function! and it's not just the usual distinction due to their use of privileged/unprivileged, it's also the fact that AIF360's definition allows the signs of the two differences to "cancel out" whereas we take the absolute value of the differences before averaging

So my suggestion would be not to create a new function, but instead add a new optional argument to our existing `equalized_odds_difference` and `equalized_odds_ratio`. For example, we could add an optional parameter called `aggregation_method` with its default value being `worst_case`, and a new possible value being `average`. The `aggregation_method` would just specify what's happening with the FPR_difference and FNR_difference -- whether we are taking the worst case or average.

[tagging @romanlutz @hildeweerts]

Reply:
I'm not too bothered by the name wrt normal usage of the word "odds" - I think we can blame the person who came up with equalized odds for that. I am very surprised that AIF360 would allow for "cancelling out" the differences... isn't that the whole point of considering the FPR and TPR separately instead of looking at overall accuracy?

I like the proposed solution @MiroDudik - in particular the usage of `worst_case` would result in less mistakes compared to `min` and `max`, I think.

Reply:
Also, if folks are in favor of just adding an optional argument to `equalized_odds_{difference,ratio}`, I'd be open to having something less verbose, e.g., we could follow pandas conventions and use `agg={worst_case,mean}`.

Reply:
I'm happy to implement Miro's suggestion. I'll get to it ASAP!

Reply:
Since @IanEisenberg couldn't continue on this I've picked it up today and continued along the lines discussed in this thread. Let me know if you have any concerns! @fairlearn/fairlearn-maintainers